packages/backend/src/db/user/user.schema.ts

```diff
@@ -18,7 +18,7 @@ export class User {
 export const UserSchema = SchemaFactory.createForClass(User);
 
 export class UserModel {
-  name: string;
+  name?: string;
```

Comment:
this should not be optional

Reply:
You have to make it optional if you want the name on CreateUserDto to be optional

Reply:
ah ok so that is typescript catching a bug that almost went into prod 😓

Reply:
the model should 100% not be optional. The controller logic is incorrect and it needs to work out the username and add it in

Reply:
i can't magic a username out of thin air

Reply:
at the moment you are just returning the name but not persisting it to the database

Reply:
you need to use @User() decorator to get the DecodedIdToken. Take a look at `CurrentUser()` method below it